dotnet/src/Connectors/Connectors.UnitTests/Memory/Redis/RedisMemoryStoreTests.cs

```diff
@@ -0,0 +1,957 @@
+﻿// Copyright (c) Microsoft. All rights reserved.
+
+using System;
+using System.Collections.Generic;
+using System.Collections.Immutable;
+using System.Linq;
+using System.Runtime.InteropServices;
+using System.Threading.Tasks;
+using Microsoft.SemanticKernel.AI.Embeddings;
+using Microsoft.SemanticKernel.AI.Embeddings.VectorOperations;
+using Microsoft.SemanticKernel.Connectors.Memory.Redis;
+using Microsoft.SemanticKernel.Memory;
+using Microsoft.SemanticKernel.Memory.Collections;
+using Moq;
+using StackExchange.Redis;
+using Xunit;
+
+namespace SemanticKernel.Connectors.UnitTests.Memory.Redis;
+
+/// <summary>
+/// Unit tests of <see cref="RedisMemoryStore"/>.
+/// </summary>
+public class RedisMemoryStoreTests
+{
+    private readonly Mock<IDatabase> _mockDatabase;
+    private readonly Dictionary<string, List<MemoryRecord>> _collections;
+
+    public RedisMemoryStoreTests()
+    {
+        this._mockDatabase = new Mock<IDatabase>();
+        this._collections = new();
+    }
+
+    private void MockCreateIndex(string collection, Action? callback = null)
```

Comment:
Please put these private mock methods to the end of test class, you can also wrap them in `#region private`. This will allow to open this test class and focus on test methods first.